### ISSUE TITLE
Medic-api-log indicating empty responses during periods of high load

### ISSUE THREAD

**Issue by kennsippell** [medic/cht-core#8279]
Looking at production medic-api-logs for a heavily loaded CHT instance,  I'm attempting to monitor the service's "yield" as defined in this [SRE Handbook](https://sre.google/sre-book/service-level-objectives/).  Basically, I'm measuring the % of requests resulting in a status code between 100 - 499. 

I'm seeing intermittent drops in yield where 1 in 3 requests are failing. Looking at requests during these periods of low yield, most of the "non-yielding requests" look like this:

```
[2023-04-17 11:45:42] RES 0a90537d-6101-4510-ace7-42714ce05b0b 197.155.150.209 - GET /medic/form%3Atb_followup_weekly/form.html HTTP/1.1 - - - ms
[2023-04-17 11:45:42] RES 0dfa85c6-2559-4c7c-b639-5032b4257ecd 197.155.150.209 - GET /medic/form%3Acontact%3Ac30_supervisor_area%3Acreate/model.xml HTTP/1.1 - - - ms
[2023-04-17 11:45:42] RES 11c38315-f93d-4fbf-ba4e-e6ca0335b866 197.155.150.209 - GET /medic/form%3Aanc_followup/form.html HTTP/1.1 - - - ms
```

The `- - -` is indicating that the response has no status code, no response time, etc.  According to [this issue](https://github.com/expressjs/morgan/issues/121) the cause of logging events like this is:

> This happens when you do not write a response and allow the app to just hang. The reason you get the dashes is because since you didn't write a response, there is no response code, there is no response time, there is no response size, etc. Basically you are seeing all the properties of the response as dashes (i.e. unknown) and the properties of the request populated in that log.

During a period of ~20 days, logs like this happen multiple times per day with the worst day having ~5000 responses like this. Here are some regex patterns of URLs which are experiencing this problem. 

url_bucket | count
-- | --
medic/form%3.* | 2400
/medic/_changes w/o longpoll | 2161
/medic/_design/medic-client | 2009
/purging/changes | 1992
/medic/_bulk_docs | 1128
/_session | 625
/dbinfo | 426
/medic/_revs_diff | 210
/medic/_all_docs | 170
medic-user-.*-meta/_local.* | 138
/medic/[a-zA-Z0-9\-]{36} | 130
/medic/login | 125
medic-user-.*-meta | 104
/api/v2/monitoring | 104
/medic/_design/medic-client/_view/doc_by_type | 103
/api/v1/users-info | 91
/api/info | 84

The majority of these responses are logged within 10 minutes, but many (~10%) are after 10-20 minutes, and many (~10%) are after 60-70 minutes.

**Comment by dianabarsan** [comment]
This looks like a timeout. 
I'm simulating this in 4.x world by: 
1. reducing the nginx request timeout to 10s
2. generating a request that exceeds this timeout

This is what I'm seeing: 
![image](https://github.com/medic/cht-core/assets/35681649/561119dd-db17-4ce0-b6a4-81662d59dcba)

While the request terminates like: 
![image](https://github.com/medic/cht-core/assets/35681649/8ff6d4b6-ff59-448b-823b-255558658588)

We used to see similar logs when we reported this issue: https://github.com/medic/cht-core/issues/7183 except I see we haven't mentioned that it looked like this in the logs.

**Comment by dianabarsan** [comment]
For this particular case, I think the request gets terminated by either exceeding the ALB timeout or by being cancelled by the client. Is there any way we can correlate the RES log entry with the REQ entry and calculate how long the request was waiting before it was terminated?

**Comment by kennsippell** [comment]
> Is there any way we can correlate the RES log entry with the REQ entry and calculate how long the request was waiting before it was terminated?

Yes, requests have a unique ID within the medic-api.log so it is easy to correlate REQ and RES. Here is [the raw data](https://docs.google.com/spreadsheets/d/1tpennYg1H5vGXK_hFTf8mfAjB2U-m3Lr8jf69PrTaJU/edit#gid=202984874) and summary.

time between REQ and RES | percent of all requests with no response
-- | --
<1sec | 16.57%
1sec | 18.95%
2 to 10sec | 33.00%
11sec to 1min | 8.24%
1min to 5min | 9.15%
5 min to 59min59sec | 7.17%
1 hour exactly (ALB) | 6.44%
>1 hour | 0.60%

> For this particular case, I think the request gets terminated by either exceeding the ALB timeout or by being cancelled by the client. 

My understanding is that the ALB idle timeout is 1 hour. Is that correct? If so, ALB would be ~7% of requests? 

I did not consider these being cancelled by the client - that's interesting.

**Comment by dianabarsan** [comment]
I'm not sure if this is a symptom of high load or not.
 
If you think this needs more attention, I would make look into % of supposedly cancelled requests vs successful requests during high and low load.

**Comment by kennsippell** [comment]
I think no action is required. Curious if you already know which behaviors will cancel requests within the client? Closing the app, deep sleep, other? I'll play around.

Thanks for sharing your knowledge.
